client/lib/fifo/fifo_unix.go

```diff
@@ -21,11 +21,13 @@ func CreateAndRead(path string) (func() (io.ReadCloser, error), error) {
 		return nil, fmt.Errorf("error creating fifo %v: %v", path, err)
 	}
 
-	openFn := func() (io.ReadCloser, error) {
+	return func() (io.ReadCloser, error) {
 		return os.OpenFile(path, unix.O_RDONLY, os.ModeNamedPipe)
-	}
+	}, nil
```

Comment:
Would it make sense to have the `unix` and `Windows` behave the same?  So `CreateAndRead` fails if the file is already present?  Or have the windows `CreateAndRead` handle if file is already present by using `winio.ListenOnlyPipe` (which may simplify the openfunction call)?

Have the subtle difference in behavior makes it notice this issue as we don't test with Windows much.

Reply:
I'll can make the unix implementation fail on file exist errors, but I'm not sure it's actually worth it. Trying to make the windows implementation behave like unix is going to bite us in too many ways.

Reply:
I guess we can call out at the documentation behavior in the doc - i.e. call out that if `CreateAndRead` behavior is undefined or platform-specific if path already exists.  But I suspect making it a well-defined consistent behavior is better though.

Reply:
FWIW we already call it out in the `godoc`, but I guess most people only see `linux|posixish|unix` docs.

Reply:
Ideally the docs for the function is the same in all files, or that we link to each other. It's quite fragile otherwise imo and a recipe for folks missing subtitles based on what files they happened to open.